src/Extensions/Diagnostics.HealthChecks/HealthCheckParameters.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	/// <summary>
+	/// Additional parameters to configure health check
+	/// </summary>
+	public class HealthCheckParameters
+	{
+		/// <summary>
+		/// Additional data to add to health check result
+		/// </summary>
+		public IReadOnlyDictionary<string, object> ReportData { get; }
+
+		/// <summary>
+		/// Creates HealthCheckParameters instance
+		/// </summary>
+		public HealthCheckParameters(KeyValuePair<string, object>[] reportData)
+		{
+			if (reportData.Length == 0)
+			{
+				ReportData = s_emptyDictionary;
+			}
+			else
+			{
+				// should be replaced by passing enumerable to constructor after we drop full framework support
+				Dictionary<string, object> dictionary = new Dictionary<string, object>(reportData.Length);
+				foreach (KeyValuePair<string, object> pair in reportData)
+				{
+					dictionary.Add(pair.Key, pair.Value);
```

Comment:
This assumes every Key in the array of KeyValuePair objects is unique and could throw.

Reply:
When fixing this, please also add UT coverage for this case.

Reply:
What would be the proper behavior for you here?
Do you think we should override values with the same key?

Reply:
Discussed in a call we'll add a more descriptive error message here.

Reply:
The discussion moved here: https://github.com/microsoft/Omex/pull/237